app/services/user_signup.rb

```diff
@@ -0,0 +1,50 @@
+class UserSignup
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+
+      charge = StripeWrapper::Charge.create({
+        amount: 999,
+        currency: "usd",
+        source: stripe_token,
+        description: "Sign up charge for #{@user.email}"
+      })
+
+      if charge.successful?
+        @user.save
+        process_invitation(invitation_token)
+        AppMailer.send_welcome_email(@user).deliver
+        @status = :success
+        self
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "Invalid user information. Please check the errors below."
+      self
+    end
```

Comment:
Since all of your if statements return `self` you could push it to the end of this method as returned value and refactor it slightly from those if branches.